### PR TITLE
VIH-10947 Add event handlers for telephone participants VIH-10949 include telephone participant ids when starting or resuming a hearing

### PR DESCRIPTION
**Pull request by shaed-parkar**
### Jira link

VIH-10947

### Change description

* Add new commands
  * AddTelephoneParticipant
  * RemoveTelephoneParticipant
  * UpdateTelephoneParticipantRoom

* Add EventHandlers
  * TelephoneJoined
  * TelephoneDisconnected
  * TelephoneTransfer

* Include telephone participant ids when starting a hearing

### CODE REVIEW

**Review by muralikaturi — APPROVED** [review]

File: VideoApi/VideoApi.UnitTests/Mappings/ConferenceCoreResponseMapperTests.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void Should_map_all_properties()
             .Excluding(x => x.Supplier)
             .Excluding(x => x.AudioRecordingRequired)
             .Excluding(x => x.MeetingRoom)
-            .Excluding(x=> x.TelephoneParticipants)
+            // .Excluding(x=> x.TelephoneParticipants)
```
Comment (will-craig) [inline VideoApi/VideoApi.UnitTests/Mappings/ConferenceCoreResponseMapperTests.cs:46]: commented code

File: VideoApi/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void Should_map_all_properties()
                 .Excluding(x => x.CaseType)
                 .Excluding(x => x.HearingVenueName)
                 .Excluding(x => x.Supplier)
-                .Excluding(x=> x.TelephoneParticipants)
+                // .Excluding(x=> x.TelephoneParticipants)
```
Comment (will-craig) [inline VideoApi/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs:52]: commented code

File: VideoApi/VideoApi.DAL/Mappings/TelephoneParticipantMap.cs [hunk 3]
```diff
@@ -10,10 +10,11 @@ public class TelephoneParticipantMap : IEntityTypeConfiguration<TelephonePartici
...
         builder.ToTable(nameof(TelephoneParticipant));
-            
+
         builder.HasKey(x => x.Id);
+        builder.Property(x => x.Id).ValueGeneratedNever();
         builder.Property(x => x.TelephoneNumber).IsRequired();
-        builder.Property(x => x.State).IsRequired().HasDefaultValue(TelephoneState.Connected);
-        builder.Property(x => x.CurrentRoom).IsRequired().HasDefaultValue(RoomType.WaitingRoom);
+        builder.Property(x => x.State).IsRequired();
+        builder.Property(x => x.CurrentRoom).IsRequired(false);
```
Comment (will-craig) [inline VideoApi/VideoApi.DAL/Mappings/TelephoneParticipantMap.cs:18]: is currentRoom not nullable by default? Does it need IsRequired(false)?

**Review by will-craig — APPROVED** [review]
Approved with some minor comments, nothing standout

File: VideoApi/VideoApi.DAL/Mappings/TelephoneParticipantMap.cs (hunk 3 above)
Comment (shaed-parkar) [reply VideoApi/VideoApi.DAL/Mappings/TelephoneParticipantMap.cs:18]: The previous migration made it required, so I need to set it to false to be nullable

File: VideoApi/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs (hunk 2 above)
Comment (shaed-parkar) [reply VideoApi/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs:52]: thanks

**Pull request merged**